src/test/java/org/opensearch/security/dlic/dlsfls/CCReplicationTest.java

```diff
@@ -0,0 +1,249 @@
+            Supplier<RepositoriesService> repositoriesServiceSupplier) {
+            if(injectedRoles != null)
+                threadPool.getThreadContext().putTransient(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROLES, injectedRoles);
+            return new ArrayList<>();
+        }
+
+        @Override
+        public List<ActionHandler<? extends ActionRequest, ? extends ActionResponse>> getActions() {
+            return Arrays.asList(new ActionHandler<>(MockReplicationAction.INSTANCE, TransportMockReplicationAction.class));
+        }
+    }
+
+    public static class MockReplicationAction extends ActionType<AcknowledgedResponse> {
+        public static final MockReplicationAction INSTANCE = new MockReplicationAction();
+        public static final String NAME = "indices:admin/plugins/replication/file_chunk";
+        private MockReplicationAction() {
+            super(NAME, AcknowledgedResponse::new);
+        }
+    }
+
+    public static class MockReplicationRequest extends AcknowledgedRequest<MockReplicationRequest> implements Replaceable {
+        private String index;
+        public MockReplicationRequest(String index) {
+            this.index = index;
+        }
+
+        public MockReplicationRequest(StreamInput inp) throws IOException {
+            super(inp);
+            index = inp.readString();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(index);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public IndicesRequest indices(String... strings) {
+            return this;
+        }
+
+        @Override
+        public String[] indices() {
+            return new String[]{index};
+        }
+
+        @Override
+        public IndicesOptions indicesOptions() {
+            return IndicesOptions.strictSingleIndexNoExpandForbidClosed();
+        }
+
+        @Override
+        public boolean includeDataStreams() {
+            return false;
+        }
+    }
+
+    public static class TransportMockReplicationAction extends HandledTransportAction<MockReplicationRequest, AcknowledgedResponse> {
+
+        @Inject
+        public TransportMockReplicationAction(TransportService transportService,
+            ActionFilters actionFilters) {
+            super(MockReplicationAction.NAME, transportService, actionFilters, MockReplicationRequest::new);
+        }
+
+        @Override
+        protected void doExecute(Task task, MockReplicationRequest request, ActionListener<AcknowledgedResponse> actionListener) {
+            actionListener.onResponse(new AcknowledgedResponse(true));
+        }
+    }
+
+    //Wait for the security plugin to load roles.
+    private void waitOrThrow(Client client, String index) throws Exception {
+        int failures = 0;
+        while(failures < 5) {
```

Comment:
Why is this explicitly set to 5?

Reply:
This is just to cut-off the loop for the test cases in case of errors.